### PR TITLE
BUG: don't allow ordering in before_trading_start

### PR DESCRIPTION
**Pull request by jbredeche**

### CODE REVIEW

File: tests/test_algorithm.py [hunk 1]
```diff
@@ -759,6 +759,48 @@ def test_invalid_order_parameters(self):
         )
         algo.run(self.data_portal)
 
+    def test_cannot_order_in_before_trading_start(self):
+        algotext = """
+from zipline.api import sid
+from zipline.errors import OrderInBeforeTradingStart
```
Comment (ehebert) [inline tests/test_algorithm.py]: I think semantically (and for easier debugging/tracebacks etc.) the `assertRaises` should be done in the test case method.

This function could be templated like this:

```
 def initialize(context):
     context.asset = sid(133)
 
 def before_trading_start(context, data):
     {order_func}(context.asset, {order_arg_1})
```

Then called with:
```
algot_text = algo_text_template.format(order_func=order_percent,  order_arg_1=1)
algo = TradingAlgorithm(script=algotext, sim_params=self.sim_params,
                                       data_frequency='daily', env=self.env)
with self.assertRaises(OrderInBeforeTradingStart):
    algo.run(self.data_portal)
```

Comment (ehebert) [inline tests/test_algorithm.py]: The second part of https://github.com/quantopian/zipline/pull/1165/commits/bc443d78b95be2f8796b201de4ed04268da0d697#r60965562 could also use `nose.parameterized`.

File: zipline/algorithm.py [hunk 2]
```diff
@@ -87,7 +88,7 @@
     require_initialized,
     require_not_initialized,
     ZiplineAPI,
-)
+    require_not_in_before_trading_start)
```
Comment (ehebert) [inline zipline/algorithm.py]: How about `disallowed_in_before_trading_start`?

Comment (jbredeche) [inline zipline/algorithm.py]: sounds good, changed

File: tests/test_algorithm.py (hunk 1 above)
Comment (jbredeche) [inline tests/test_algorithm.py]: good idea, changed

**Pull request merged**